PacketWrapper/src/main/java/com/comphenix/packetwrapper/WrapperPlayServerEntityEquipment.java

```diff
@@ -61,49 +64,116 @@ public void setEntityID(int value) {
 	}
 
 	/**
-	 * Retrieve the entity of the painting that will be spawned.
+	 * Retrieve the entity whose equipment will be changed.
 	 * 
 	 * @param world - the current world of the entity.
-	 * @return The spawned entity.
+	 * @return The affected entity.
 	 */
 	public Entity getEntity(World world) {
 		return handle.getEntityModifier(world).read(0);
 	}
 
 	/**
-	 * Retrieve the entity of the painting that will be spawned.
+	 * Retrieve the entity whose equipment will be changed.
 	 * 
 	 * @param event - the packet event.
-	 * @return The spawned entity.
+	 * @return The affected entity.
 	 */
 	public Entity getEntity(PacketEvent event) {
 		return getEntity(event.getPlayer().getWorld());
 	}
 
+        /**
+         * Retrieve list of ItemSlot - ItemStack pairs.
+         *
+         * @return The current Item
+         */
+        public List<Pair<ItemSlot, ItemStack>> getSlotStackPairs() {
+            return handle.getSlotStackPairLists().read(0);
+        }
+
+        /**
+         * Set a ItemSlot - ItemStack pair.
+         * @param slot The slot the item will be equipped in. If matches an existing pair, will overwrite the old one
+         * @param item The item to equip
+         */
+        public void setSlotStackPair(ItemSlot slot, ItemStack item) {
+            List<Pair<ItemSlot, ItemStack>> slotStackPairs = handle.getSlotStackPairLists().read(0);
+            slotStackPairs.removeIf(pair -> pair.getFirst().equals(slot));
+            slotStackPairs.add(new Pair<>(slot, item));
```

Comment:
You need to write the slot stack pair list after updating, at least when creating this packet from scratch.
```java
handle.getSlotStackPairLists().write(0, slotStackPairs);
```

Reply:
Same goes for when you're modifying an existing packet